llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -921,6 +950,46 @@ static std::string getESIMDIntrinSuffix(id::FunctionEncoding *FE,
   return Suff;
 }
 
+static void translateBlockLoad(CallInst &CI, bool IsSLM) {
+  IRBuilder<> Builder(&CI);
+
+  APInt Val =
+      parseTemplateArg(CI, 2, ESIMDIntrinDesc::GenXArgConversion::TO_I64);
+  MaybeAlign Align(Val.getZExtValue());
+
+  auto Op0 = CI.getArgOperand(0);
+  auto DataType = CI.getType();
+  if (IsSLM) {
+    // Convert 'uint32_t' to 'addrspace(3)*' pointer.xxyy
+    auto PtrType = PointerType::get(DataType, 3);
+    Op0 = Builder.CreateIntToPtr(Op0, PtrType);
+  }
+
+  auto LI = Builder.CreateAlignedLoad(DataType, Op0, Align, CI.getName());
+  LI->setDebugLoc(CI.getDebugLoc());
+  CI.replaceAllUsesWith(LI);
+}
+
+static void translateBlockStore(CallInst &CI, bool IsSLM) {
+  IRBuilder<> Builder(&CI);
+
+  APInt Val =
+      parseTemplateArg(CI, 2, ESIMDIntrinDesc::GenXArgConversion::TO_I64);
```

Comment:
same idea

Reply:
Fixed in 1e722d1af21d4d979f9e352e053a0ae4d340e45a